app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/documents/DocumentsFragment.kt

```diff
@@ -23,7 +27,17 @@ class DocumentsFragment : Fragment(R.layout.contract_detail_documents_fragment)
             applyNavigationBarInsets()
             val documentsAdapter = DocumentAdapter(trackingFacade)
             adapter = documentsAdapter
-            model.documentsList.observe(viewLifecycleOwner, documentsAdapter::submitList)
+            model.viewState
+                .flowWithLifecycle(lifecycle)
+                .onEach { viewState ->
+                    when (viewState) {
+                        ContractDetailViewModel.ViewState.Error -> {}
+                        ContractDetailViewModel.ViewState.Loading -> {}
+                        is ContractDetailViewModel.ViewState.Success -> {
+                            documentsAdapter.submitList(viewState.state.documentsViewState.documents)
+                        }
+                    }
+                }.launchIn(viewLifecycleScope)
```

Comment:
I think the previous implementation meant that in case the documentsList was empty, the empty list was submitted to the documentsAdapter.
With the new implementation, if somehow we first get a success with a nonEmpty list of documents it will be submitted, and later a Loading/Error state will not change those already existing documents.
I am missing some context here, but is this possible? If it's not, this is okay as it is. If this is possible, are we okay with this interaction? If yes, then again it's no problem. If however we don't want to do that, we may want to add `documentsAdapter.submitList(emptyList())` in the Error/Loading states